lib/cache/role.go

```diff
@@ -130,6 +130,17 @@ func (c *Cache) ListRoles(ctx context.Context, req *proto.ListRolesRequest) (*pr
 	return &resp, nil
 }
 
+// ListRequestableRoles is a paginated requestable role getter.
+// This method delegates to the auth service and does not use the cache, it is here only to satisfy the auth.Cache interface.
+func (c *Cache) ListRequestableRoles(ctx context.Context, req *proto.ListRequestableRolesRequest) (*proto.ListRequestableRolesResponse, error) {
+	ctx, span := c.Tracer.Start(ctx, "cache/ListRequestableRoles")
+	defer span.End()
+
+	// Always delegate to auth service since requestable roles depend on user context and cannot be cached
+	resp, err := c.Config.Access.ListRequestableRoles(ctx, req)
+	return resp, trace.Wrap(err)
+}
+
```

Comment:
I don't think this is sound.  This will end up listing the requestable roles _of the teleport service_, not of the caller.  I don't think we can allow caches to implement this method.